hash_practice/exercises.py

```diff
@@ -2,18 +2,61 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(nlogn) - beacuse of the sort
+        Space Complexity: O(n)
     """
-    pass
+    repeated_words = {}
+    for word in strings:
+        sorted_word = "".join(sorted(word))
+        if sorted_word not in repeated_words:
+            repeated_words[sorted_word] = [word]
+        else:
+            repeated_words[sorted_word].append(word)
+    
+    anagrams_list = []
+    for words in repeated_words.values():
+        anagrams_list.append(words)
+    
+    return anagrams_list
+
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 Well done, I would say that the time complexity is O(nk) due to the final nested loops in the function.